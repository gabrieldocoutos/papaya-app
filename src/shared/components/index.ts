import { Text } from "./Text/TextComponent";
import { View, SafeAreaView } from "./View/ViewComponent";

export { Text, View, SafeAreaView };
