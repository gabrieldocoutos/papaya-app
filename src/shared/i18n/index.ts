import homeScreen from "./homeScreen.json";

export { homeScreen };
