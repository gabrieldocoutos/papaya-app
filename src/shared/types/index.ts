import type { RootStackParamList } from "./navigatorTypes";
export { RootStackParamList };
