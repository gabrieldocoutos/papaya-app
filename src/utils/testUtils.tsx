import React, { FC, ReactElement } from "react";
import { render, RenderOptions } from "@testing-library/react-native";
import { ThemeProvider } from "styled-components/native";
import { theme } from "shared/configuration";

const AllTheProviders: FC = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, "wrapper">) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react-native";
export { customRender as render };
