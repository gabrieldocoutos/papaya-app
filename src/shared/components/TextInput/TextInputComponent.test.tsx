import React from "react";
import { render } from "utils/testUtils";

import { TextInput } from "shared/components";

test("snapshot with regular fontWeight", () => {
  const { toJSON } = render(<TextInput label="E-mail" color="primary" />);
  expect(toJSON()).toMatchSnapshot();
});
