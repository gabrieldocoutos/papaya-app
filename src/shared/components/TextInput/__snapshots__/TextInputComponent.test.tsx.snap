// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot with regular fontWeight 1`] = `
<View
  color="primary"
  flexDirection="column"
  style={
    Array [
      Object {
        "color": "#FF6243",
        "flexDirection": "column",
      },
    ]
  }
>
  <Text
    color="primary"
    fontFamily="Lato_700Bold"
    fontSize={16}
    mb="3px"
    style={
      Array [
        Object {
          "color": "#FF6243",
          "fontFamily": "Lato_700Bold",
          "fontSize": 16,
          "marginBottom": 3,
        },
      ]
    }
  >
    E-mail
  </Text>
  <View
    borderRadius="5px"
    borderWidth={1}
    color="primary"
    height={44}
    paddingLeft={16}
    style={
      Object {
        "borderColor": "rgba(238, 238, 238, 1)",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "#FF6243",
        "flexDirection": "row",
        "height": 44,
        "paddingLeft": 16,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      color="secondary"
      onBlur={[Function]}
      onFocus={[Function]}
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "color": "#2f2e41",
            "width": "100%",
          },
        ]
      }
      underlineColorAndroid="transparent"
    />
  </View>
</View>
`;
